runtime/src/accounts.rs

```diff
@@ -362,8 +362,7 @@ impl Accounts {
                 {
                     if let Some(program) = self
                         .accounts_db
-                        .load(ancestors, &programdata_address)
-                        .map(|(account, _)| account)
+                        .load_and_keep_in_read_only_cache(ancestors, &programdata_address)
```

Comment:
@sakridge, is it possible for the read only cache to have an incorrect version of its contents due to an account coming from a different ancestors?

Reply:
Yes. The cache would need to take the ancestors into account.

Reply:
it does now